### DISCUSSION TITLE
How to retrieve whether the project is snapshot with Gradle Groovy DSL?

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#810] [Q&A] [answered]
Hi,

I am trying to retrieve via a Gradle property whether the project is snapshot. I can't seem to find how to do so.

I would like to set `releaseName` depending on that property, something like: `releaseName = jreleaser.project.snapshot ? 'Release early-access' : 'Release {{tagName}}'` but that doesn't work.

**Comment by aalmiray** [comment, marked as answer]
Hmm what you're looking for may be a bit tricky as the snapshot status is determined during model evaluation which happens after Gradle's configuration phase. Also, `jreleaser.project.snapshot` is a domain object, not a boolean or a value that may be turned into a boolean using Groovy Truth (coincidentally  Groovy Truth will always return `true` as the reference is never null).

https://github.com/jreleaser/jreleaser/blob/f7d97b0e2275e5dc96096303e05bd09b71023916/plugins/jreleaser-gradle-plugin/src/main/groovy/org/jreleaser/gradle/plugin/dsl/Project.groovy#L64-L80

In the meantime you'll have to apply your own logic to find out if the project is snapshot or not. By default this is done by matching the `jreleaser.project.snapshot.pattern` regex to the project's version. The regex' default value is `.*-SNAPSHOT`.

**Reply by gotson** [reply]
> Hmm what you're looking for may be a bit tricky as the snapshot status is determined during model evaluation which happens after Gradle's configuration phase

Makes sense !

---

Solved using the following:

```groovy
releaseName = version ==~ '.*-SNAPSHOT' ? 'Release early-access' : 'Release {{tagName}}'
overwrite = version ==~ '.*-SNAPSHOT'
```

**Reply by aalmiray** [reply]
FWIW "Release early-access" is the default name when a project is deemed as snapshot, while "Release {{tagName}}" is the default when a project is non snapshot. Also, `overwrite` is set to `true` when snapshot. Thus your customizations in this case match what JReleaser already does under the covers 😏 

You can verify this by running `greadle jreleaserConfig` with a snapshot and non-snapshot version.

**Reply by gotson** [reply]
Thanks, i was using `jreleaserConfig` to check, somehow i missed that `overwrite` would switch to true, but i couldn't clearly see how `releaseName` changed because it always reference `{{tagName}}` which doesn't seem to change in the output.

I always get the following output, whatever if it's SNAPSHOT or not, so that confused me:

```
tagName: v{{projectVersion}}
releaseName: Release {{tagName}}
```

**Reply by aalmiray** [reply]
Ah yes, true. The value of `{{tagName}}` stays the same but the value of `jreleaser.project.snapshot.label` (which is "early-access" by default) will be used instead as I believe updating the model such that `tagName = 'early-access'` caused an issue. I may need to revisit this decision.
